src/lib.rs

```diff
@@ -84,12 +85,10 @@ impl<T> MiniVec<T> {
     }
 
     fn data(&self) -> *mut T {
-        if self.buf_.is_null() {
-            core::ptr::null_mut()
-        } else {
-            let count = get_offset::<T>();
-            unsafe { self.buf_.add(count) as *mut T }
-        }
+        debug_assert!(!self.buf_.is_null());
```

Comment:
should you allocate instead? I don't think std's Vec panic when there is no elements

Reply:
nevermind im stupid

Reply:
Ha ha, no such thing!

Asking questions is a good thing and it seems like you've answered it yourself.